packages/neos-ui-extensibility/src/registry/SynchronousRegistry.js

```diff
@@ -1,38 +1,74 @@
 import AbstractRegistry from './AbstractRegistry';
+import {positionalArraySorter} from '@neos-project/utils-helpers';
 
 export default class SynchronousRegistry extends AbstractRegistry {
     constructor(description) {
         super(description);
 
-        // internal registry, containing the keys as object keys, and the values being the values in the registry.
-        this._registry = {};
-        this._keys = [];
+        this._registry = [];
     }
 
-    add(key, value) {
-        // TODO: implement position in registry!
-        // TODO: "key" must be string!
-        this._registry[key] = value;
-        this._keys.push(key);
+    set(key, value, position = 0) {
+        if (typeof key !== 'string') {
+            throw new Error('Key must be a string');
+        }
+        if (typeof position !== 'string' && typeof position !== 'number') {
+            throw new Error('Position must be a string or a number');
+        }
+        const entry = {key, value};
+        if (position) {
+            entry.position = position;
+        }
+        this._registry.push(entry);
 
-        return this._registry[key];
+        return value;
     }
 
     get(key) {
-        // TODO: "key" must be string!
-        return this._registry[key];
+        if (typeof key !== 'string') {
+            throw new Error('Key must be a string');
+        }
+        const result = this._registry.find(item => item.key === key);
+        return result ? result.value : false;
+    }
+
+    _getChildrenWrapped(searchKey) {
+        const unsortedChildren = this._registry.filter(item => item.key.indexOf(searchKey + '/') === 0);
+        return positionalArraySorter(unsortedChildren);
+    }
+
+    getChildrenAsObject(searchKey) {
+        const result = {};
+        this._getChildrenWrapped(searchKey).forEach(item => {
+            result[item.key] = item.value;
+        });
+        return result;
+    }
+
+    getChildren(searchKey) {
+        return this._getChildrenWrapped(searchKey).map(item => item.value);
     }
 
     has(key) {
```

Comment:
can you add docs to all the public methods here, how they shall be used?

Reply:
Will do.